client/response.go

```diff
@@ -15,13 +15,23 @@ type Response struct {
 }
 
 func (r *Response) Unmarshal(v interface{}) error {
+	defer r.Close()
+
 	if err := json.NewDecoder(r.Body).Decode(&v); err != nil {
 		return fmt.Errorf("failed to json decode read bytes: %w", err)
 	}
 
 	return nil
 }
 
+// Close reads all of the body stream and closes it to make sure that tcp connections can be reused properly
+func (r *Response) Close() error {
+	io.Copy(io.Discard, r.Body) //nolint: errcheck
```

Comment:
Could we return these two errors? Even if most people won't check them.

Reply:
We could but I dont know how to handle error from copy. I don't think that we want no prevent close due to an error in copy?

Reply:
I agree that we should close on `io.Copy` errors. Perhaps something like this?
```go
func (r *Response) Close() error {
  if err := io.Copy(io.Discard, r.Body); err != nil {
    r.Body.Close()
    
    return err
  }

  return r.Body.Close()
}

```